lib/vector-buffers/src/buffer_usage_data.rs

```diff
@@ -98,31 +110,26 @@ pub struct BufferUsageData {
     received_byte_size: AtomicU64,
     sent_event_count: AtomicU64,
     sent_byte_size: AtomicU64,
-    dropped_event_data: Option<BufferUsageDroppedEventData>,
+    dropped_event_count: AtomicU64,
+    dropped_event_byte_size: AtomicU64,
+    dropped_event_count_intentional: AtomicU64,
+    dropped_event_byte_size_intentional: AtomicU64,
```

Comment:
We've split out intentionally vs unintentionally dropped events here to better adhere to the Buffer specification. although admittedly, it does make the code a little denser. My thought in the future would be to try and bundle up the event count/event byte size atomics into some helper struct and then there would just be one field per category... sent, received, intentionally dropped, unintentionally dropped.

Still sort of just hiding things and spreading out the complexity, but a lot easier to read, more likely.